File: Dogs-C/Dogs-C/View Controllers/JLImageCollectionViewController.m
//
//  JLImageCollectionViewController.m
//  Dogs-C
//
//  Created by Jordan Lamb on 10/9/19.
//  Copyright © 2019 Squanto Inc. All rights reserved.
//

#import "JLImageCollectionViewController.h"

@interface JLImageCollectionViewController ()

@end

@implementation JLImageCollectionViewController

static NSString * const reuseIdentifier = @"imageCell";

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = self.breed.name.capitalizedString;
    if(self.subBreed){
        NSString *title = [[self.subBreed.name.capitalizedString stringByAppendingString:@" "]
                           stringByAppendingString:self.breed.name.capitalizedString];
        self.title = title;
    }
    [self fetchImageURLs];
}


#pragma mark <UICollectionViewDataSource>
- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
    return 1;
}


- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return self.imageURLs.count;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    JLImageCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:reuseIdentifier forIndexPath:indexPath];
    NSURL *imageURL = [NSURL URLWithString:self.imageURLs[indexPath.row]];
    cell.imageURL = imageURL;
    return cell;
}

- (void)fetchImageURLs
{
     if(self.subBreed)
     {
         [[JLBreedController sharedInstance] fetchSubBreedImageURLs: self.subBreed breed: self.breed completion:^(NSArray *fetchedURLs) {
             self.imageURLs = fetchedURLs;
             dispatch_async(dispatch_get_main_queue(), ^{
                 [[self collectionView] reloadData];
             });
         }];
     } else {
         [[JLBreedController sharedInstance] fetchBreedImageURLs:self.breed completion:^(NSArray *fetchedURLs) {
             self.imageURLs = fetchedURLs;
             dispatch_async(dispatch_get_main_queue(), ^{
                 [[self collectionView] reloadData];
             });
         }];
     }
}

#pragma mark - Navigation

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if([segue.identifier isEqualToString:@"toDetailVC"])
    {
        NSIndexPath *indexPath = [[self.collectionView indexPathsForSelectedItems]firstObject];
        NSURL *dogImageURL = [NSURL URLWithString:self.imageURLs[indexPath.row]];
        JLDetailViewController *destinationVC = segue.destinationViewController;
        destinationVC.dogImageURL = dogImageURL;
        if(self.breed){
            destinationVC.breed = self.breed;
        }
        if(self.subBreed){
            destinationVC.subBreed = self.subBreed;
        }
    }
}

@end

File: Dogs-C/Dogs-C/View Controllers/JLImageCollectionViewController.swift
//
//  JLImageCollectionViewController.swift
//  Dogs-C
//
//  Created by Jordan Lamb on 10/9/19.
//  Copyright © 2019 Squanto Inc. All rights reserved.
//

import UIKit

private let reuseIdentifier = "imageCell"

class JLImageCollectionViewController: UICollectionViewController {
    
    var breeds: JLBreed?

    override func viewDidLoad() {
        super.viewDidLoad()
        
    }


    // MARK: - Navigation

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
    }

    // MARK: UICollectionViewDataSource

    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return 0
    }

    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath)
    
        return cell
    }

    // MARK: UICollectionViewDelegate


    // Uncomment this method to specify if the specified item should be selected
    override func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
        return true
    }

}
